src/libraries/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.cs

```diff
@@ -103,6 +103,11 @@ public Utf8JsonWriter(IBufferWriter<byte> bufferWriter, JsonWriterOptions option
         {
             _output = bufferWriter ?? throw new ArgumentNullException(nameof(bufferWriter));
             _options = options;
+
+            if (_options.MaxDepth == 0)
```

Comment:
Is it verified anywhere that MaxDepth of `1` and `2` works consistently for read and write? i.e. the reader and writer have the same semantics. Also, depending on the viewpoint, a depth of `1` may not imply any object\array support.

Reply:
Yes. Ultimately, the semantics stem from the semantics of the `Utf8JsonWriter/Reader.CurrentDepth` properties, which are consistent and we already have testing for those:

1. https://github.com/dotnet/runtime/blob/0749730dbbe7e9815d99839c3b9333ea409ba98f/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonReaderTests.cs
2. https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonWriterTests.cs

The meaning of `MaxDepth` being set to `0` is special, and in `JsonWriterOptions` it specifically mirrors the semantics of `Utf8JsonReader`.